File: Programmation/TD2/flexion-verbe.rb
#!/usr/bin/env ruby

myPronoms = ["je", "tu", "il/elle", "nous", "vous", "ils/elles"]
myTerminaisons = ["e", "es", "e", "ons", "ez", "ent"]

puts "Entrez un verbe du 1er groupe : "
verbe = gets.chomp
radical = verbe.gsub(/er$/, '')

puts "Les formes du verbe " + verbe + " au présent de l'indicatif sont :\n"


(0..myPronoms.length).each{|i| puts myPronoms[i] + " " + radical + myTerminaisons[i]}

File: Programmation/TD2/flexion-verbe.pl
#!/usr/bin/perl -w
use strict ;

my ( $radical, $inf, $verb, $i ) ;
my @pronoms = ( "je", "tu", "il/elle", "nous", "vous", "ils/elles" ) ;
my @terminaisons = ( "e", "es", "e", "ons", "ez", "ent" ) ;

print "Entrez un verbe du 1er groupe : " ;
chomp($verb=<STDIN>) ;
if ($verb =~ m/er$/){
    $inf = $verb ;
    $inf =~ s/er$//;
    
    
    print "Les formes du verbe \" $verb \" au présent de l'indicatif sont :\n" ;
    
    for ($i=0; $i<@pronoms; $i++)
    {
	print $pronoms[$i] . " " . $inf . $terminaisons[$i] . "\n" ;
    }
}

else{print "Ceci n'est pas un verbe du premier groupe !\n"}
